### PR TITLE
New features

### PR DESCRIPTION
**Pull request by lotusnoir**
adding new features for S3 buckets

### CODE REVIEW

File: README.md
Comment (benoit-garcia) [inline README.md:129]: Could you move this example to a dedicated sub-folder? You can refer to Hashicorp [Standard Module Structure](https://developer.hashicorp.com/terraform/language/modules/develop/structure) for this.

File: main.tf
Comment (benoit-garcia) [inline main.tf:1]: Ok to renamed, not adding a "moved" block will break users. Should be:
```suggestion
moved {
  from = scaleway_object_bucket.this
  to   = scaleway_object_bucket.main
}

resource "scaleway_object_bucket" "main" {
```

File: main.tf
Comment (benoit-garcia) [inline main.tf:50]: Same than before: Add a "moved" block. Should be:
```suggestion
moved {
  from = scaleway_object_bucket_lock_configuration.this
  to   = scaleway_object_bucket_lock_configuration.main
}

resource "scaleway_object_bucket_lock_configuration" "main" {
```

File: versions.tf
Comment (benoit-garcia) [inline versions.tf]: I'm not sure there's a lot of changes on bucket related resources since v2.10.0. Should be:
```suggestion
      version = ">= 2.10.0"
```

File: versions.tf
Comment (benoit-garcia) [inline versions.tf]:
```suggestion
  required_version = ">= 1.3"
```

**Review by benoit-garcia — CHANGES_REQUESTED** [review]
Hello,
Thanks for your contribution!
There's a few things that could be improved before merging, I made some suggestions.
Let me know if I can help with them.

File: README.md
Comment (lotusnoir) [reply README.md:129]: ok i'll make a separate commit to not block this one

**Comment by jgalais** [comment]
Hello,

Thank you @lotusnoir  and @benoit-garcia for your job.

@benoit-garcia Can you validate this Pull request ?

I want to create S3 bucket but tag usage it's an important thing.
I try to avoid to fork all Scaleway module to avoid to recreate the wheel each time.

Regards,

**Comment by benoit-garcia** [comment]
Hello,

@jgalais 
I've added support of tags on [v2.0.0](https://github.com/scaleway-terraform-modules/terraform-scaleway-bucket/releases/tag/v0.2.0). Feel free to test and let me know.

@lotusnoir 
Thanks for you work. I'd like to maintain some consistency on code style.
As I couldn't edit your pull request, I create #34 based on your code and improved/cleaned it a little.
Hope you won't mind; your contributions are highly appreciated.

**Comment by jgalais** [comment]
Hello,

I tested this morning and tags function is ok.

Thank you.

Regards,

**Pull request closed without merging**
